packages/office-ui-fabric-react/src/components/ResizeGroup/ResizeGroup.base.tsx

```diff
@@ -186,6 +187,46 @@ export const getNextResizeGroupStateProvider = (measurementCache = getMeasuremen
     };
   }
 
+  /**
+   * Scale the given data based on the measurements already in cache and based on the current container dimension
+   * This is useful to optimize changes to ResizeGroup props which do not affect scaling. In that case,
+   * we could avoid measuring the resize group.
+   * @param data - The initial data point to start measuring.
+   * @param onReduceData - Function that transforms the data into something that should render with less width/height.
+   * @param onGrowData - Function that transforms the data into something that may take up more space when rendering.
+   */
+  function scaleDataBasedOnMeasurementsInCache(data: any, onReduceData: (prevData: any) => any, onGrowData?: (prevData: any) => any): any {
+    let scaledData = { ...data };
+    let measuredDimension;
+    let grow = true;
+    do {
+      measuredDimension = _measurementCache.getCachedMeasurement(scaledData);
+      if (measuredDimension === undefined) {
+        return { ...data };
+      }
+
+      const nextScaledData;
+      if (grow && onGrowData && measuredDimension < _containerDimension!) {
+        nextScaledData = onGrowData(scaledData);
+      } else if (measuredDimension > _containerDimension!) {
+        if (grow) {
+          grow = false;
+        }
+        nextScaledData = onReduceData(scaledData);
+      } else {
+        break;
```

Comment:
Add a comment as to why this is a termination condition

Reply:
I think I also made this more simpler....let me know how you like it now !